bindings/js/api.js

```diff
@@ -7,18 +7,31 @@ if (typeof module !== 'undefined') {
   var shell = require(path.resolve(__dirname, './shell.js'));
   var api_promise = Promise.resolve(require(path.resolve(__dirname, './api.json')));
   var layouts_promise = Promise.resolve(require(path.resolve(__dirname, './layouts.json')));
-  var modules_promise = Promise.resolve(shell.my_modules.map(m => {
-    var source = fs.readFileSync(path.resolve(__dirname, './' + m + ".wasm"));
-    return new Uint8Array(source);
-  }));
 
 } else {
   var loader = this;
   var shell = this;
   var api_promise = fetch("api.json").then(r => r.json());
   var layouts_promise = fetch("layouts.json").then(r => r.json());
-  var modules_promise = Promise.all(my_modules.map(m => fetch(m + ".wasm")))
-    .then(response => Promise.all(response.map(r => r.arrayBuffer())));
+}
+
+// We now allow the user to pass a custom list of modules if they want to do
+// their own, more lightweight packaging.
+function getModulesPromise(modules) {
+  if (typeof module !== 'undefined')
+    return Promise.resolve((modules || shell.my_modules).map(m => {
+      var source = fs.readFileSync(path.resolve(__dirname, './' + m + ".wasm"));
+      return ({
+        buf: new Uint8Array(source),
+        name: m
+      });
+    }));
+  else
+    return Promise.all((modules || my_modules)
+      .map(name =>
+        fetch(name + ".wasm")
+        .then(r => r.arrayBuffer())
+        .then(buf => ({ buf, name }))));
```

Comment:
Why not factorize the code a bit by extracting the "in browser / in module" behavior, for instance like that:
```js
function getModulesPromise(modules = shell.my_modules) {
    const readModule = async path =>
        typeof module !== 'undefined'
            ? new Uint8Array(await fs.readFile(path.resolve(__dirname, './' + path)))
            : (await fetch(path)).arrayBuffer();
    return Promise.all(modules.map(async name =>
        ({ buf: await readModule(name + ".wasm"), name })
    ));
}
```